tests/testthat/test_other-resource-types.R

```diff
@@ -0,0 +1,94 @@
+test_that("test call_sedt_api works with non-file resources", {
+  library(gpkg)
+  nc = sf::read_sf(system.file("shape/nc.shp", package="sf"))
+  nc_path <- system.file("shape/nc.shp", package="sf")
+  nc = sf::read_sf(nc_path)
+  nc_points <- suppressWarnings(sf::st_centroid(nc))
```

Comment:
why centroid and not `point_on_surface()`. It's probably okay but worth checking.

Reply:
good point - I updated to `point_on_surface()`

Reply:
(pun not intended)